### PR TITLE
V3 compatible, Recalc improved, progress bars fixed 

### LINKED ISSUE #276: 2021 (v3) scheduler support
MorphMan can skip new cards that meet certain criteria (see under MorphMan’s settings → *General* → *Review Preferences*). This works with the v2 scheduler, but not with v3: it outputs messages such as `%d alternatives will be skipped`, but no cards are actually skipped.

-----

I tried to modify Morphman’s code in order to add v3 support. The following lines wrap `my_fillNew` around the vanilla `_fillNew`:

https://github.com/kaegi/MorphMan/blob/3fde85f31ad1c51b1095d4824d7474ca69d9fcba/morph/newMorphHelper.py#L83-L84

([here](https://github.com/ankitects/anki/blob/dc80804af6a2dfd6900fb61f30fbe63b9f767311/pylib/anki/scheduler/v2.py#L159-L189) are lines that define `_fillNew` for the v2 scheduler in Anki 2.1.49)

The problem is, the v3 scheduler doesn’t have a `_fillNew` (see [here](https://github.com/ankitects/anki/blob/2.1.49/pylib/anki/scheduler/v3.py)). This means MorphMan has to tackle v3 scheduler support differently. The question is how…

### PR DESCRIPTION
**Pull request by mortii**
- Now works on the V3 scheduler (fixes #276)
- Exponentially faster recalc action
- Progress dialogs are now shown when recalcing and anki no longer freezes (background thread is used instead of main thread)
- Renamed some files that were poorly named
- Refactored init file and extracted hidden hooks to make the code execution logic much clearer
- Added some type hints for clearer logic
- refactored some variable names to follow PEP conventions

### CODE REVIEW

**Review by landonepps — CHANGES_REQUESTED** [review]
Overall looks good, I appreciate the attention to detail like renaming variables.
I'm happy to merge it if you can address the linter warnings.

Also, the `master` branch is outdated. The latest version on AnkiWeb is using the `qt6-migration` branch. I'll merge your changes into that branch after it gets merged to master.

File: morph/main.py
Comment (mortii) [reply morph/main.py:150]: aah, nice!

**Comment by mortii** [comment]
I fixed the lint errors and I rebased onto qt6-migration

**Comment by mortii** [comment]
hold on, a very a sneaky error appeared that needs to be fixed

**Comment by mortii** [comment]
there was an invisible input parameter after removing a lambda, but it's fixed now.

**Comment by mortii** [comment]
Also this should probably be placed in the ankiweb description until we get a proper wiki/guide (next on my todo-lis):

> NB! If the deck you are studying has sub-decks then new cards will by default only be gathered from the first sub-deck until it is empty before looking for new cards in the next sub-deck. If you instead want to get new i+1 cards from all sub-decks do the following:
> 1. Activate the v3 scheduler: Tools -> Review -> Scheduler -> V3 scheduler
> 2. Deck that has sub-decks: Deck options -> Display Order -> New card gather order -> Ascending position

The writing is kinda clunky, feel free to improve it in whatever way you see fit , lol.

**Comment by landonepps** [comment]
Looks good to me, thanks for all the work you did here.
And I appreciate the AnkiWeb note, I'll be sure to add it.

**Comment by PrentissLiu** [comment]
Thank you so much for your work. I am the version of QT5. I tried to modify: 
readability_ui.py and readability_settings_ui.py : 

> from PyQt6 import QtCore, QtGui, QtWidgets

to:

> from PyQt5 import QtCore, QtGui, QtWidgets

it works!

**Comment by landonepps** [comment]
Awesome, I really appreciate the report!
Both of my computers are arm-based Macs, which don't support Qt5, making it quite annoying to test.

If this is really all that's needed, I should be able to modify the UI generation code to replace these imports with the aqt version and have full backwards compatibility.

**Pull request merged**
